strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticator.java

```diff
@@ -0,0 +1,29 @@
+package org.carlspring.strongbox.authentication.api.impl.ldap;
+
+import org.carlspring.strongbox.authentication.api.Authenticator;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class LdapAuthenticator
```

Comment:
Things are separated now. [Authentication suppliers](https://github.com/strongbox/strongbox/blob/master/strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/suppliers/AuthenticationSupplier.java) and [Authenticators](https://github.com/strongbox/strongbox/blob/master/strongbox-authentication-api/src/main/java/org/carlspring/strongbox/authentication/api/Authenticator.java). 

[Authentication suppliers](https://github.com/strongbox/strongbox/blob/master/strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/suppliers/AuthenticationSupplier.java) give us an Authentication instance (right now most likely UsernamePasswordAuthenticationToken) where we have user credentials from. Currently we support 2 authentication suppliers: HTTP Basic and JWT. Both supply Authentication (user credentials).

Then we have separated set of  [Authenticators](https://github.com/strongbox/strongbox/blob/master/strongbox-authentication-api/src/main/java/org/carlspring/strongbox/authentication/api/Authenticator.java). These things can be provided by strongbox clients (see https://github.com/strongbox/strongbox/wiki/How-To-Implement-A-Custom-Authenticator for reference) but we also have couple of built-in [Authenticators](https://github.com/strongbox/strongbox/blob/master/strongbox-authentication-api/src/main/java/org/carlspring/strongbox/authentication/api/Authenticator.java):
* [Default authenticator](https://github.com/strongbox/strongbox/blob/master/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/DefaultAuthenticator.java) with [DaoAuthenticationProvider](https://github.com/spring-projects/spring-security/blob/master/core/src/main/java/org/springframework/security/authentication/dao/DaoAuthenticationProvider.java) and [StrongboxUserDetailService](https://github.com/strongbox/strongbox/blob/master/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/userdetails/StrongboxUserDetailService.java)
* [LDAP authenticator ](https://github.com/fuss86/strongbox/blob/SB-129/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticator.java) with [LdapAuthenticationProvider](https://github.com/spring-projects/spring-security/blob/master/ldap/src/main/java/org/springframework/security/ldap/authentication/LdapAuthenticationProvider.java) (where we check given authentication to) and [DefaultLdapAuthoritiesPopulator](https://github.com/fuss86/strongbox/blob/SB-129/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/ldap/DefaultLdapAuthoritiesPopulator.java) (where we map LDAP roles to internal roles ([see an example](https://github.com/fuss86/strongbox/blob/SB-129/strongbox-user-management/src/test/resources/etc/conf/strongbox-authentication-providers.xml#L46)))
* Active Directory authenticator (in the near future)
* CAS authenticator (in the future)
* .... (in the future)

Above things are joined together in [StrongboxAuthenticationFilter](https://github.com/strongbox/strongbox/blob/master/strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/StrongboxAuthenticationFilter.java#L49).

> So is there any examples of how LDAP can be used with JWT?

There are no examples right now. If you deliver JWT token in the appropriate HTTP header then user credentials will be taken from it using [JWTAuthenticationSupplier](https://github.com/strongbox/strongbox/blob/master/strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/suppliers/JWTAuthenticationSupplier.java). Then the authentication instance will be used against  [LDAP authenticator ](https://github.com/fuss86/strongbox/blob/SB-129/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/api/impl/ldap/LdapAuthenticator.java).